### PR TITLE
Pinned doc deps

### PR DESCRIPTION
**Pull request by MichaelHatherly**
As requested by @tkelman, pinned versions of documentation dependencies. Also reinstates the stacktraces stdlib file that had been left out of the index.

### CODE REVIEW

File: doc/REQUIRE [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Compat 0.9.5 0.9.5+
+DocStringExtensions 0.3.0 0.3.0+
+Documenter 0.8.2 0.8.2+
```
Comment (MichaelHatherly) [inline doc/REQUIRE]: @tkelman, I assume you did mean pinning *all* the dependencies rather than just Documenter?

Comment (tkelman) [inline doc/REQUIRE]: right

**Review by tkelman — APPROVED** [review]
> their content is a duplicate of what is found in the nav menu.

How does that nav menu get populated with an index then?

**Comment by MichaelHatherly** [comment]
> How does that nav menu get populated with an index then?

Those two index pages contain the same set of page links as the two subsections in https://github.com/JuliaLang/julia/blob/4dcada105716dc30c8e7edcc843c7d8dcc9bffac/doc/make.jl#L87-L110. (`PAGES` is what is used to build the nav menu.)

**Comment by MichaelHatherly** [comment]
Also includes fixes for https://github.com/JuliaLang/julia/pull/18588#commitcomment-20126414 and https://github.com/JuliaLang/julia/pull/18588#discussion_r91622605.

The last commit reformats `devdocs/ast.md` quite a bit to get a more consistent layout, which I think is more readable and consistent.

**Pull request merged**